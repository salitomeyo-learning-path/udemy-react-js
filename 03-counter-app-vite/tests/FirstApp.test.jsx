// import { render } from '@testing-library/react';
// import { FirstApp } from '../src/FirstApp';
// import React from "react";


// describe('Pruebas en <FirstApp />', () => {

//     test('debe de hacer match con el snapshot', () => {
//         const title = "Hola, Soy Goku";

//         const { container } = render( <FirstApp title={title} /> );

//         expect( container ).toMatchSnapshot();
//     });

//     test('debe de mostrar el titulo en un h1', () => {
//         const title = "Hola, Soy Goku";
        
//         const { container, getByText } = render( <FirstApp title={title} /> );

//         expect( getByText(title) ).toBeTruthy();

//         const h1 = container.querySelector('h1');
//         expect( h1.innerHTML ).toBe(title);
//     })

//     test('debe mostrar el elemento by test id', () => {
//         const title = "Hola, Soy Goku";
        
//         const { container, getByTestId } = render( <FirstApp title={title} /> );

//         expect( getByTestId('kitty') ).toBeTruthy();
//         expect( getByTestId('kitty').innerHTML ).toContain('Jade');
//     })

//     test('debe mostrar el subtitulo enviado por prop', () => {
//         const title = "Hola, Soy Goku";
//         const subTitle = "Soy un subtitulo";
        
//         const { container, getByText } = render( 
//         <FirstApp 
//             title={title}
//             subTitle={ subTitle}
//         /> );

//         expect( getByText(subTitle) ).toBeTruthy();
//     })

// })