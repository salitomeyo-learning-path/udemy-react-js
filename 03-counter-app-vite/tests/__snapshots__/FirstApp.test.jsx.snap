// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <h4>
    No hay subtitulo
  </h4>
  <p
    data-testid="kitty"
  >
    Jade
  </p>
</div>
`;
